retrieval/raw/raw_endpoint.go

```diff
@@ -114,13 +123,121 @@ func downloadToFile(c *Client, method, sourceName string, workDir *os.File, URL
 		return filename, errors.New("Unable to create raw metric file")
 	}
 	defer util.SafeClose(rawRespFile.Close, &rerr)
-
 	filename = rawRespFile.Name()
 
+	if _, ok := ParsableFileSet[sourceName]; c.parseMetricData && ok {
+		err = parseAndWriteData(sourceName, resp.Body, rawRespFile)
+		return filename, err
+	}
+
 	_, err = io.Copy(rawRespFile, resp.Body)
 	if err != nil {
 		return filename, errors.New("Error writing file: " + rawRespFile.Name())
 	}
 
 	return filename, rerr
 }
+
+// TODO: investigate streamed json reading / writing
+func parseAndWriteData(filename string, reader io.Reader, writer io.Writer) error {
+	var to = getType(filename)
+	out := reflect.New(reflect.TypeOf(to))
+	err := json.NewDecoder(reader).Decode(out.Interface())
+
+	if err != nil {
+		return err
+	}
+	to = sanitizeData(out.Elem().Interface())
+
+	data, err := json.Marshal(to)
+	if err != nil {
+		return err
+	}
+	_, err = io.Copy(writer, bytes.NewReader(data))
+	return err
+}
+
+func getType(filename string) interface{} {
+	var to interface{}
+	switch filename {
+	case Nodes:
+		to = NodeList{}
+	case Namespaces:
+		to = NamespaceList{}
+	case Pods:
+		to = PodList{}
+	case PersistentVolumes:
+		to = PersistentVolumeList{}
+	case PersistentVolumeClaims:
+		to = PersistentVolumeClaimList{}
+	case AgentMeasurement:
+		to = CldyAgent{}
+	case Services, ReplicationControllers:
+		to = LabelMapMatchedResourceList{}
+	case Deployments, ReplicaSets, Jobs, DaemonSets:
+		to = LabelSelectorMatchedResourceList{}
+	}
+	return to
+}
+
+func sanitizeData(to interface{}) interface{} {
+	switch to.(type) {
+	case LabelSelectorMatchedResourceList:
+		return sanitizeSelectorMatchedResourceList(to)
+	case PodList:
+		return sanitizePodList(to)
+	case LabelMapMatchedResourceList:
+		return sanitizeMapMatchedResourceList(to)
+	}
+	return to
+}
+
+func sanitizeSelectorMatchedResourceList(to interface{}) interface{} {
+	cast := to.(LabelSelectorMatchedResourceList)
+	for i, item := range cast.Items {
+
+		// stripping env var and related data from the object
+		item.ObjectMeta.ManagedFields = nil
+		if _, ok := item.ObjectMeta.Annotations[KubernetesLastAppliedConfig]; ok {
+			delete(item.ObjectMeta.Annotations, KubernetesLastAppliedConfig)
+		}
+		cast.Items[i] = item
+	}
+	return cast
+}
+
+func sanitizePodList(to interface{}) interface{} {
+	cast := to.(PodList)
+	for i, item := range cast.Items {
+
+		// stripping env var and related data from the object
+		item.ObjectMeta.ManagedFields = nil
+		if _, ok := item.ObjectMeta.Annotations[KubernetesLastAppliedConfig]; ok {
+			delete(item.ObjectMeta.Annotations, KubernetesLastAppliedConfig)
+		}
+		for j, container := range item.Spec.Containers {
+			container.Env = nil
+			item.Spec.Containers[j] = container
```

Comment:
It shouldn't only filter "env" variable. Instead, it should only extract/keep fields that needs by Containers Insight.
The task also includes identifying the fields need be kept.
For Pods, following fields should be kept in my opinion, but it needs be verified with Denormizer:
* name
* namespace
* node
* labels
* status
containers[]:
* id
* state
* limits
* requests

Reply:
Hey Peter. The method of serialization/deserialization used here does remove a decent amount of non-relevant data in addition to the env specific filtering done above by using the structs we also use internally for parsing this data. This ticket accomplishes the primary task as requested by the customer as well as provides the ability to easily decrease our data collected in the future.

While it would be nice to do a full audit of the data used by our internal processes and pare down to 
*exactly* what we need, I do not believe that is a reasonable goal at this moment. I would be happy to discuss the specific challenges of this internally if you'd like.